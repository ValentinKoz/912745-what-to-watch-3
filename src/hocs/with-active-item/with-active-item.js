import React, {PureComponent} from "react";
import PropTypes from "prop-types";

export const withActiveItem = (Component) => {
  class WithActiveItem extends PureComponent {
    constructor(props) {
      super(props);

      this.state = {
        activeItem: null
      };

      this.handleSetActiveItem = this.handleSetActiveItem.bind(this);
    }

    handleSetActiveItem(film) {
      this.setState({activeItem: film});
    }

    render() {
      const {activeItem} = this.state;

      return (
        <Component
          {...this.props}
          onSetActiveItem={this.handleSetActiveItem}
          activeItem={activeItem}
        />
      );
    }
  }
  WithActiveItem.propTypes = {
    films: PropTypes.array.isRequired,
    onCardClickHandle: PropTypes.func,
  };

  return WithActiveItem;
};
