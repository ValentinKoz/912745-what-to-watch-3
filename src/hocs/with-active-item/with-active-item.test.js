import React from "react";
import renderer from "react-test-renderer";
import {withActiveItem} from "./with-active-item.js";

const films = [{
  genre: `genre`,
  title: `title`,
  releaseDate: 1000,
  poster: `img/the-grand-budapest-hotel-poster.jpg`,
  bgPoster: `img/moonrise-kingdom.jpg`,
  ratingScore: `0,0`,
  ratingLevel: `level`,
  ratingCount: 1,
  text: `...text`,
  director: `director`,
  runTime: `0h 00m`,
  starring: `starring`,
  video: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  reviews: [{
    text: `...text`,
    author: `author`,
    time: `December 24, 2016`,
    rating: `0,0`
  }, {
    text: `...text`,
    author: `author`,
    time: `December 24, 2016`,
    rating: `0,0`
  }]
}, {
  genre: `genre`,
  title: `title`,
  releaseDate: 1000,
  poster: `img/the-grand-budapest-hotel-poster.jpg`,
  bgPoster: `img/moonrise-kingdom.jpg`,
  ratingScore: `0,0`,
  ratingLevel: `level`,
  ratingCount: 1,
  text: `...text`,
  director: `director`,
  runTime: `0h 00m`,
  starring: `starring`,
  video: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  reviews: [{
    text: `...text`,
    author: `author`,
    time: `December 24, 2016`,
    rating: `0,0`
  }, {
    text: `...text`,
    author: `author`,
    time: `December 24, 2016`,
    rating: `0,0`
  }]
}];

const MockComponent = () => <div></div>;
const MockComponentWrapped = withActiveItem(MockComponent);

it(`withActiveItem is renderer correctly`, () => {
  const tree = renderer
    .create(
        <MockComponentWrapped
          films={films}
          onCardClickHandle={() => {}}
        />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
