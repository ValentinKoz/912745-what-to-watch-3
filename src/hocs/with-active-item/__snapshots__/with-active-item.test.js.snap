// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withActiveItem is renderer correctly 1`] = `<div />`;
